File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    mid = (sum(queue1) + sum(queue2)) // 2
    sum1 = sum(queue1)

    queue1 = deque(queue1)
    queue2 = deque(queue2)

    while queue1 and queue2:
        if sum1 > mid:
            cur = queue1.popleft()
            sum1 -= cur

        elif sum1 < mid:
            cur = queue2.popleft()
            sum1 += cur
            queue1.append(cur)
        else: return answer
        answer += 1

    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    let answer = 0
    const max = queue1.length * 2
    let sum = queue1.reduce((a,c,i)=> a + c - queue2[i],0)/2
    let [i,j] = [0,0]
    while (sum !== 0 && answer < max+3){
        if (sum > 0){
            const v = queue1[i]
            i++
            sum -= v
            queue2.push(v)
        } else {
            const v = queue2[j]
            j++
            sum += v
            queue1.push(v)
        }
        answer++
    }
    
    return sum !== 0 ? -1 : answer
}